File: TeamService/src/main/java/kr/lovesignal/teamservice/model/response/Member.java
package kr.lovesignal.teamservice.model.response;

import kr.lovesignal.teamservice.entity.MemberEntity;
import lombok.*;

import java.time.LocalDate;
import java.time.Period;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Member {
    private String nickname;
    private int age;
    private String description;
    private String profileImage;
    private String memberUUID;
    private boolean teamLeader;

    public static List<Member> buildMembers(List<MemberEntity> memberEntities){
        List<Member> members = new ArrayList<>();
        for(MemberEntity memberEntity : memberEntities){
            LocalDate birthDate = LocalDate.parse(memberEntity.getBirth(), DateTimeFormatter.BASIC_ISO_DATE);
            int age = Period.between(birthDate, LocalDate.now()).getYears();
                    Member member = Member.builder()
                    .nickname(memberEntity.getNickname())
                    .age(age)
                    .memberUUID(memberEntity.getUUID().toString())
                    .description(memberEntity.getDescription())
                    .teamLeader("T".equals(memberEntity.getTeamLeader()) ? true : false)
                    .build();
            members.add(member);
        }
        return members;
    }
}

File: frontend/love_signal/src/types/member.ts
export type team = {
  teamUUID: string;
  members: member[];
};

export type member = {
  nickname: string;
  age: number;
  description: string;
  profileImage: string;
};

//회원정보 조회시에 받아오는 데이터(굳이 필요한가 의문)
export type userInfo = {
  memberUUID: string;
  nickname: string;
  gender: string;
  age: number;
  description: string;
  teamUUID: string;
  teamLeader: boolean;
};
